src/functions.rs

```diff
@@ -699,6 +699,28 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_get_aux_type_checking() {
+        let db = Connection::open_in_memory().unwrap();
+        db.create_scalar_function("example", 2, false, |ctx| {
+            if !ctx.get::<bool>(1)? {
+                ctx.set_aux::<i64>(0, 100);
+            } else {
+                assert_eq!(ctx.get_aux::<String>(0), Err(Error::GetAuxWrongType));
+                assert_eq!(ctx.get_aux::<i64>(0), Ok(Some(&100)));
+            }
+            Ok(true)
+        })
+        .unwrap();
+
+        let res: bool = db.query_row(
+            "WITH foo(i) AS (VALUES (0), (1)) SELECT example(0, i) FROM foo",
```

Comment:
[WITH clause](https://sqlite.org/releaselog/3_8_3.html) are only supported by SQLite >= 3.8.3.
I suggest you use this syntax instead to fix CI builds:
```sql
SELECT example(0, i) FROM (SELECT 0 as i UNION SELECT 1);
```